### DISCUSSION TITLE
Hidden Check Box Modal Control "Legacy"? Going away?

### DISCUSSION THREAD

**Discussion by Fish1** [saadeghi/daisyui#2220] [General]
Is the hidden checkbox control for the modal control going away?

Correct me if anything I say is wrong...

But the new modal controls via javascript and the ID of the modal are very difficult to use and maintain. The ID of my modal isn't tracked via my LSP, so eslint is yelling at me that my modal ID doesn't exist.

The "legacy" checkbox system allowed me to store and track a boolean with my LSP, and then pass that variable to the checkbox. So it was easy for me to track the open/close state of my modal & control it with a single variable. With the new imperative JavaScript method, I lose that ability and must remember to run side effects every time I open a modal.

I'd really like if something like the "legacy" system was still maintained...

**Comment by saadeghi** [comment]
It's not going to be removed anytime soon, don't worry.  

But it is recommended to use the new HTML `<dialog>` element because it has better accessibility. It locks the keyboard navigation inside the modal, it won't have z-index issues because it's on a top layer, it works better with screen readers and there's no need to put the modal at the end do the page. The need to use JS might be annoying but that's just how `<dialog>` element works.

**Reply by wuarmin** [reply]
@saadeghi ok, thank you. But what's the best way to use the new HTML `<dialog>` element, and close or open the modal by a modalState? Using `useEffect` like this?:
```js
const dialogRef = useRef<HTMLDialogElement>(null)

useEffect(() => {
    if (dialogRef.current?.open && !showModal.show) {
      dialogRef.current?.close()
    } else if (!dialogRef.current?.open && showModal.show) {
      dialogRef.current?.showModal()
    }
}, [showModal.show])

// ...

return (
   <dialog ref={dialogRef}>
    ...
   </dialog>
)
```
Thanks!

**Reply by saadeghi** [reply]
I don't use React anymore so I don't know if it's currently the best way or not.  

This might be helpful:  
https://dev.to/elsyng/react-modal-dialog-using-html-dialog-element-5afk
Or this  
https://medium.com/@dimterion/modals-with-html-dialog-element-in-javascript-and-react-fb23c885d62e
